src/services/XeroService.ts

```diff
@@ -0,0 +1,100 @@
+import { Inject, Injectable } from "@nestjs/common";
+import { ConfigService } from "@nestjs/config";
+import { Logger } from "ethers/lib/utils";
+import { Repository } from "typeorm";
+import { Request } from "../data/entities/request.entity";
+import { Contact, XeroClient, LineItem, Invoice, Invoices } from "xero-node";
+import {
+    ConfigSettings,
+    IXeroService,
+    XeroTokenSet,
+    XeroClients
+} from "../interfaces";
+
+const XERO_SCOPES =
+    "openid profile email accounting.transactions accounting.contacts offline_access";
+const XERO_INVOICE_DESCRIPTION = "KYC";
+
+@Injectable()
+export class XeroService implements IXeroService {
+    private client: XeroClient;
+    private readonly logger = new Logger("XeroService");
+    private accountCode;
+    private price;
+    private tenantId;
+    private gpibId;
+
+    constructor(
+        private config: ConfigService,
+        @Inject("REQUEST_REPOSITORY")
+        private requestRepository: Repository<Request>
+    ) {
+        this.client = new XeroClient({
+            clientId: this.config.get(ConfigSettings.XERO_CLIENT_ID) ?? "",
+            clientSecret:
+                this.config.get(ConfigSettings.XERO_CLIENT_SECRET) ?? "",
+            scopes: XERO_SCOPES.split(" ")
+        });
+        this.accountCode =
+            this.config.get(ConfigSettings.XERO_SALES_CODE) ?? "";
+        this.price = this.config.get(ConfigSettings.XERO_PRICE) ?? "";
+        this.tenantId = this.config.get(ConfigSettings.XERO_TENANT_ID) ?? "";
+        this.gpibId = this.config.get(ConfigSettings.XERO_GPIB_ID) ?? "";
+    }
+
+    public async sendInvoices(authToken: XeroTokenSet): Promise<string> {
+        try {
+            // set the auth token from the POST request body
+            this.client.setTokenSet(authToken);
+
+            // create initial xero constants that won't change
+            const contact: Contact = {
+                contactID: this.gpibId
```

Comment:
let's not hardcode gpibId here, contactID should pass in from controller

Reply:
👍